### ISSUE TITLE
Variables not read from GUI (version 1.20)

### ISSUE THREAD

**Issue by tlaloc911** [usebruno/bruno#2556]
### I have checked the following:

- [X] I use the newest version of bruno.
- [X] I've searched existing issues and found nothing related to my issue.

### Describe the bug

In latest version, variables defined in GUI at request level are not recognized, neither bru.getVar(), nor variable inspector

### Screenshots/Live demo link

Before

![image](https://github.com/usebruno/bruno/assets/29415755/857ad9ae-ddb7-4085-a359-56127c8b4059)

![image](https://github.com/usebruno/bruno/assets/29415755/72e9a463-bf5b-4942-a657-17665e2d3f8b)

After run: 

![image](https://github.com/usebruno/bruno/assets/29415755/c6f89510-8a4c-459a-9c0d-733a81caa312)

Note that Var1 doesn't appear, also it cannot be retreived with getVar()

![image](https://github.com/usebruno/bruno/assets/29415755/b6127737-4099-4429-98d2-177194d45883)

**Comment by SPIRIT-00** [comment]
Hi there, it's a requst level var now. Use `bru.getRequestVar()`, but `bru.setRequestVar()` not exist.

**Comment by anubhavhazarika** [comment]
Hi, @helloanoop, Hi, @lohxt1, 

We notice that the 'feature' of bru.getRequestVar() was added in [v1.20.0](https://github.com/usebruno/bruno/releases/tag/v1.20.0) via https://github.com/usebruno/bruno/pull/2541 . 

As we are unable to find any other reference/documentation for this enhancement, could you please help us better understand the intended behavior of this feature? 
Is it intended to **replace** getVar(), specific to the request variables (i.e. variables defined in the 'Vars' tab in the UI)? If so, this will be a major change and not backward compatible with the earlier expected behavior of request variables (https://docs.usebruno.com/scripting/vars - 'Vars are available **both at collection and request level**.').

We already have a large number of test cases written that rely on getVar() to read the variables set in the request (as they get persisted as collection variables prior to v1.20.0) and after this change, everything is currently breaking as getVar() returns null for request variables. 

As reported above, maybe the issue is related to the persistence of the request variables, that is to say, variables defined at the request level still should be persisted as collection variables (as before, prior to v1.20.0) and they can be read using getVar() and getRequestVar()?

**Comment by neutc** [comment]
Yes, the newly added method getRequestVar() is not described in nowhere, and the functionality of getVar() is changed without backward compatible, it will cause many confusions for beginner to learn Bruno.

**Comment by Thexumaker** [comment]
why is a breaking change in a minor update?

**Comment by helloanoop** [comment]
Apologies for the confusion here.

We are working on Collection Variables that will have collection level scope. This is being planned to be released in 2 weeks.

The behaviour where the scope of request vars being leaked to the collection scope was a _bug_. There were many other tickets where this behaviour was causing confusion.

> We already have a large number of test cases written that rely on getVar() to read the variables set in the request

@anubhavhazarika Would it be possible to use `bru.getRequestVar()`. This should get you access to the value of the request variable inside the script/test during the request execution.

If anyone needs help in fixing the test suite due to this change, I'd be happy to get on a call and help: https://cal.com/helloanoop/30min

**Comment by fortion71** [comment]
> why is such a huge a breaking change in a minor update and not well documented. Will also add for some reason getRequestVar works albeit it isn't documented anywhere, but we also still have to use setVar. setRequestVar doesn't work...

Please improve the versioning policies to avoid this in the future!

I paid for the Golden edition to support the devs, since I was under the impression this was as high-quality of a project as it seemed. But this was 20min of digging to find out that v1.20 was causing my script to not work. Seems like it was an easily avoidable mistake – either don't include breaking changes in a minor version (or find another way to version this thing), or at least document breaking changes in release notes (at the bare minimum) or docs. Alternatively (or better yet, additionally), have a Canary branch or similar where users will expect breaking changes.

(P.S. Other than this, I've been enjoying Bruno, and wish you all the best success 😃) @helloanoop

**Comment by tlaloc911** [comment]
Closing this after understand new behaviour and API functions
